app/src/main/java/com/github/orkest/Model/Authorization.kt

```diff
@@ -1,56 +1,108 @@
 package com.github.orkest.Model
 
-import okhttp3.FormBody
-import okhttp3.OkHttpClient
-import okhttp3.Request
-import org.json.JSONObject
-import okhttp3.*
-import java.io.IOException
-
-import java.util.concurrent.CompletableFuture
+import android.app.Activity
+import android.content.Intent
+import android.util.Base64
+import com.spotify.sdk.android.auth.AuthorizationClient
+import com.spotify.sdk.android.auth.AuthorizationRequest
+import com.spotify.sdk.android.auth.AuthorizationResponse
+import java.security.MessageDigest
 
 
+/**
+ * This class is used to get the access token from the Spotify API
+ *
+ * It implements methods for the authorization code flow with PKCE Flow. It
+ * is the recommended authorization flow if you’re implementing authorization
+ * in a mobile app, single page web app, or any other type of application where
+ * the client secret can’t be safely stored
+ */
 class Authorization {
 
     companion object{
 
-        fun getAccessToken(): String? {
-            val client_id = Providers.SPOTIFY.CLIENT_ID // Your client id
-            val client_secret = "1fc2f06625b8415ea071cf3cf46d390b" // Your secret
-
-            val client = OkHttpClient()
-            val requestBody = FormBody.Builder()
-                .add("grant_type", "client_credentials")
-                .build()
-            val request = client_id?.let { Credentials.basic(it, client_secret) }?.let {
-                Request.Builder()
-                    .url("https://accounts.spotify.com/api/token")
-                    .addHeader("Authorization", it)
-                    .post(requestBody)
+        private val CODE_VERIFIER = generateRandomString()
+
+        /**
+         * Code verifier
+         *
+         * @return a random string with a length between 43 and 128 characters
+         */
+        fun generateRandomString(): String {
+            val allowedChars = ('A'..'Z') + ('a'..'z') + ('0'..'9')
+            return (1..128)
+                .map { allowedChars.random() }
+                .joinToString("")
+        }
+
```

Comment:
How are you ensuring that the string length is between 43 and 128 characters ? It seems like the string length is always 128